lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -3900,6 +3938,45 @@ void XorRPrimOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
   genericAsmResultNames(*this, setNameFn);
 }
 
+//===----------------------------------------------------------------------===//
+// Cross Module Reference verifiers.
+//===----------------------------------------------------------------------===//
+
+LogicalResult RefSendOp::verify() {
+  // Check that the flows make sense.
+  if (failed(checkConnectFlow(*this)))
+    return failure();
+
+  return success();
+}
+
+LogicalResult RefRecvOp::verify() {
+  // Check that the flows make sense.
+  if (failed(checkConnectFlow(*this, true)))
+    return failure();
+  // The result of the ref recv op (xmr write) cannot be used as the destination
+  // of any connect op in the module.
+  Value res = getResult();
+  if (res.hasOneUse())
+    return success();
+  for (auto u : res.getUsers())
+    if (auto cLike = dyn_cast<FConnectLike>(u)) {
+      if (cLike.getDest() != res)
+        continue;
+      auto resRef = getFieldRefFromValue(res);
+      bool rootKnown;
+      auto resName = getFieldName(resRef, rootKnown);
+      auto diag = emitError();
+      diag << "ref recv (xmr write) result ";
+      if (rootKnown)
+        diag << "\"" << resName << "\" ";
+      diag << "cannot be used as the destination of a connect";
+      return diag.attachNote(cLike.getLoc()) << "the connect was defined here";
```

Comment:
This check logic is re-used in the flow checks above. Can this be factored into a static function?